src/replication.c

```diff
@@ -911,10 +911,16 @@ static void appendFollowerCb(struct raft_io_append *req, int status)
         }
     }
 
-    if (r->state != RAFT_FOLLOWER) {
-        tracef("local server is not follower -> don't send result");
+    /* If our term number has changed since receiving these entries, current_leader
+     * might have changed as well, so don't send a response. */
+    if (r->current_term != args->term) {
```

Comment:
A node can move to `RAFT_CANDIDATE` without updating its term due to the pre-vote mechanism see
https://github.com/canonical/raft/blob/425a6b57bdea8a600402eac6ee1ab1ea97b36215/src/election.c#L122 

So probably we need something more like `r->current_term != args->term || r->state != RAFT_FOLLOWER`